05_classify_mutations_multimodal/plot_multimodal_results.ipynb

```diff
@@ -845,12 +848,12 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "EGFR diff: -0.0286 (pval: 0.2479)\n",
-      "IDH1 diff: 0.0040 (pval: 0.8424)\n",
-      "KRAS diff: -0.0013 (pval: 0.9424)\n",
-      "PIK3CA diff: -0.0054 (pval: 0.8047)\n",
-      "SETD2 diff: -0.0070 (pval: 0.8165)\n",
-      "TP53 diff: -0.0037 (pval: 0.7065)\n"
+      "EGFR diff: -0.0286 (pval: 0.1484)\n",
+      "IDH1 diff: 0.0040 (pval: 0.8438)\n",
+      "KRAS diff: -0.0013 (pval: 0.8438)\n",
+      "PIK3CA diff: -0.0054 (pval: 0.6406)\n",
+      "SETD2 diff: -0.0070 (pval: 0.6406)\n",
+      "TP53 diff: -0.0037 (pval: 0.6406)\n"
```

Comment:
I probably don't understand because I'm a garbage statistician, but is there a reason that you see the same p-values across tests? Is it because it's a ranked test?

Reply:
It is because it's a ranked test - that means there's only a discrete number of possibilities for test statistics/p-values, and since there's only n=8 for these tests the odds of seeing the same p-value for 2 comparisons are fairly high.

Just to verify, I wrote a quick Python script to calculate the test statistics by hand for the duplicated p-values here, using [the description on Wikipedia](https://en.wikipedia.org/wiki/Wilcoxon_signed-rank_test#Test_procedure):

<img width="259" alt="Screen Shot 2021-07-12 at 1 25 21 PM" src="https://user-images.githubusercontent.com/2345877/125330353-a579b500-e314-11eb-877c-d7a9111fd489.png">

(the distribution of the test statistic is symmetric so +/- 4 result in the same p-value). So the repeated p-values do have the same test statistic, since the rank * sign combinations end up being equivalent.